Loading `robot_description` from topic does not work as expected

**Describe the bug**
There are two ways to load the robot description in the controller manager 1) via a parameter 2) via a topic. If I try to load the description via a topic, I get an error:
```
 [ros2_control_node-1] [WARN] [1703336900.604603840] [resource_manager]: (hardware 'PandaFakeSystem'): 'panda_joint1/position' command interface not in available list. This should not happen (hint: multiple cleanup calls).
``` 
The reason this happens is because `ControllerManager::init_resource_manager` does not get called before the first call to `ControllerManager::read`. This is because it takes some time to receive robot description from the topic. 

**To Reproduce**
It should be reproducible in most configurations as long as the robot description is published to a topic instead of passed in as a parameter. 

**Expected behavior**
I expect the controller manager to either wait for the resource manager to be initialized before entering the main loop or maybe wait to load controllers until the resource manager is initialized. 

**Environment (please complete the following information):**
 - OS: Ubuntu 22.04
 - Version Humble

Attempted to implement loading `robot_description` from a topic instead of a param in moveit studio and never see the robot render. 
Found this in the logs:
```
moveit_studio-drivers-1       | [ros2_control_node-2] [ERROR] [1704303580.172077753] [joint_state_broadcaster]: None of requested interfaces exist. Controller will not run.
moveit_studio-drivers-1       | [ros2_control_node-2] [WARN] [1704303580.172122053] [joint_state_broadcaster]: Error occurred while doing error handling.
moveit_studio-drivers-1       | [ros2_control_node-2] [ERROR] [1704303580.172147027] [controller_manager]: After activation, controller 'joint_state_broadcaster' is in state 'unconfigured' (1), expected 'active' (3).
moveit_studio-drivers-1       | [ros2 run controller_manager spawner --controller-manager-timeout 180 joint_state_broadcaster-5] [INFO] [1704303580.174731442] [spawner_joint_state_broadcaster]: Configured and activated joint_state_broadcaster
```
And when listing the controllers I see the joint_state_broadcaster controller is `unconfigured`:
```
dwyackzan@dwy-workstation:~/user_ws$ ros2 control list_controllers
joint_trajectory_controller[joint_trajectory_controller/JointTrajectoryController] inactive  
io_and_status_controller[ur_controllers/GPIOController] inactive  
robotiq_gripper_controller[position_controllers/GripperActionController] inactive  
force_torque_sensor_broadcaster[force_torque_sensor_broadcaster/ForceTorqueSensorBroadcaster] inactive  
robotiq_activation_controller[robotiq_controllers/RobotiqActivationController] inactive  
streaming_controller[position_controllers/JointGroupPositionController] inactive  
joint_trajectory_controller_chained_open_door[joint_trajectory_controller/JointTrajectoryController] inactive  
joint_state_broadcaster[joint_state_broadcaster/JointStateBroadcaster] unconfigured
```
@destogl could you help us figure out a solution and provide some feedback on [Paul's fix](https://github.com/pac48/ros2_control/commit/a1bc50cfbd418627eb7a36a5d265f1aed3fa8097)

--- a/controller_manager/src/controller_manager.cpp
+++ b/controller_manager/src/controller_manager.cpp
@@ -313,6 +313,10 @@ ControllerManager::ControllerManager(
     RCLCPP_WARN(get_logger(), "'update_rate' parameter not set, using default value.");
   }
 
+  if (resource_manager_->is_urdf_already_loaded())
+  {
+    init_services();
+  }
   subscribe_to_robot_description_topic();
 
   diagnostics_updater_.setHardwareID("ros2_control");

--- a/controller_manager/src/controller_manager.cpp
+++ b/controller_manager/src/controller_manager.cpp
@@ -286,12 +286,12 @@ ControllerManager::ControllerManager(
       "[Deprecated] Passing the robot description parameter directly to the control_manager node "
       "is deprecated. Use '~/robot_description' topic from 'robot_state_publisher' instead.");
     init_resource_manager(robot_description_);
+    init_services();
   }
 
   diagnostics_updater_.setHardwareID("ros2_control");
   diagnostics_updater_.add(
     "Controllers Activity", this, &ControllerManager::controller_activity_diagnostic_callback);
-  init_services();
 }
 
 ControllerManager::ControllerManager(
@@ -318,7 +318,6 @@ ControllerManager::ControllerManager(
   diagnostics_updater_.setHardwareID("ros2_control");
   diagnostics_updater_.add(
     "Controllers Activity", this, &ControllerManager::controller_activity_diagnostic_callback);
-  init_services();
 }
 
 void ControllerManager::subscribe_to_robot_description_topic()
@@ -352,6 +351,7 @@ void ControllerManager::robot_description_callback(const std_msgs::msg::String &
       return;
     }
     init_resource_manager(robot_description_);
+    init_services();
   }
   catch (std::runtime_error & e)
   {

--- a/controller_manager/test/test_hardware_management_srvs.cpp
+++ b/controller_manager/test/test_hardware_management_srvs.cpp
@@ -84,7 +84,9 @@ public:
         "Unable to initialize resource manager, no robot description found.");
     }
 
-    cm_->init_resource_manager(robot_description);
+    auto msg = std_msgs::msg::String();
+    msg.data = robot_description_;
+    cm_->robot_description_callback(msg);
 
     SetUpSrvsCMExecutor();
   }
@@ -383,7 +385,9 @@ public:
         "Unable to initialize resource manager, no robot description found.");
     }
 
-    cm_->init_resource_manager(robot_description);
+    auto msg = std_msgs::msg::String();
+    msg.data = robot_description_;
+    cm_->robot_description_callback(msg);
 
     SetUpSrvsCMExecutor();
   }
@@ -440,7 +444,9 @@ public:
         "Unable to initialize resource manager, no robot description found.");
     }
 
-    cm_->init_resource_manager(robot_description);
+    auto msg = std_msgs::msg::String();
+    msg.data = robot_description_;
+    cm_->robot_description_callback(msg);
 
     SetUpSrvsCMExecutor();
   }

